### ISSUE TITLE
Child Themes: template-parts links should include the child theme slug and not the parent theme slug.

### ISSUE THREAD

**Issue by matiasbenedetto** [WordPress/create-block-theme#275]
When a template part is linked from a pattern it should have the `theme` key and it should contain the theme slug of the child theme.

Example:
```
<!-- wp:template-part {"slug":"header","theme":"childthemeslug"} /-->
```

Currently, when we export a child theme we use the parent theme slug and that avoids the template part to be loaded as expected in the editor causing the error: _Template part has been deleted or is unavailable_.

More info in this Gutenberg issue: https://github.com/WordPress/gutenberg/issues/44243 particularly in https://github.com/WordPress/gutenberg/issues/44243#issuecomment-1470414943

**Comment by miksansegundo** [comment]
Just noting that the same error `Template part has been deleted or is unavailable` reported in this issue also happens on themes that are exported with the "Clone" option. 

<img width="374" alt="Screenshot 2566-03-20 at 10 37 06" src="https://user-images.githubusercontent.com/1881481/226240893-ea64bcb7-c262-4d99-b71d-20af16434995.png">

Updating the value of the `theme` key in the `template-part` to use the cloned or current theme slug didn't help in that case.

**Comment by matiasbenedetto** [comment]
> Updating the value of the theme key in the template-part to use the cloned or current theme slug didn't help in that case.

@miksansegundo I'm interested in this you mentioned. Did you try adding `"theme":"yourthemeslug"` ?

Could you attach the zip file obtained to help to debug?

**Comment by miksansegundo** [comment]
@matiasbenedetto here are the steps I followed:

- Install and activate `blank-canvas-3` theme
- From the editor add a `Front Page` template 
     - Add a `Header` template part
     - Edit the template part
- Export the theme using `CBT plugin` with `Clone` option using the name `Lodestar`. See [theme test.zip](https://github.com/WordPress/create-block-theme/files/11036099/theme.test.zip)  

Notice that CBT plugin exported a front-page pattern with the header template part using the value `lodestar-3-wpcom` in the theme attribute, which was unexpected: 
`<!-- wp:template-part {"slug":"header","theme":"lodestar-3-wpcom","tagName":"header"} /-->`.

When I upload and activate the theme on a Jurassic site, I got the same error as in this issue. See screenshot:

<img width="774" alt="Screenshot 2566-03-22 at 11 13 26" src="https://user-images.githubusercontent.com/1881481/226800518-acde351a-a8d1-42f8-92d5-3910efae0584.png">

After updating the theme attribute to be `"theme":"lodestar"` or `"theme":"blank-canvas-3"` the error persisted.
